model/simulationtests/daylighting_devices.rb

```diff
@@ -0,0 +1,151 @@
+dome.setName('Dome')
+dome.setSubSurfaceType('TubularDaylightDome')
+dome.setSurface(window1.surface.get)
+
+points = OpenStudio::Point3dVector.new
+points << OpenStudio::Point3d.new(0, 1, 0.1)
+points << OpenStudio::Point3d.new(0, 0, 0.1)
+points << OpenStudio::Point3d.new(1, 0, 0.1)
+points << OpenStudio::Point3d.new(1, 1, 0.1)
+diffuser = OpenStudio::Model::SubSurface.new(points, model)
+diffuser.setName('Diffuser')
+diffuser.setSubSurfaceType('TubularDaylightDiffuser')
+diffuser.setSurface(window1.surface.get)
+
+tubular = OpenStudio::Model::DaylightingDeviceTubular.new(dome, diffuser, construction)
+tubular.setDiameter(1.1)
+tubular.setTotalLength(1.4)
+tubular.setEffectiveThermalResistance(0.28)
+
+transition_zone = OpenStudio::Model::TransitionZone.new(thermal_zone, 1)
+tubular.addTransitionZone(transition_zone)
+
+###############################################################################
+#                         DaylightingDeviceLightWell                          #
+###############################################################################
+
+sub_surface = window1
+
+light_well = OpenStudio::Model::DaylightingDeviceLightWell.new(sub_surface)
+light_well.setHeightofWell(1.2)
+light_well.setPerimeterofBottomofWell(12.0)
+light_well.setAreaofBottomofWell(9.0)
+light_well.setVisibleReflectanceofWellWalls(0.7)
+
+###############################################################################
+#                           DaylightingDeviceShelf                            #
+###############################################################################
+
+sub_surface = window2
+
+#### Inside shelf
+partitionGroup = OpenStudio::Model::InteriorPartitionSurfaceGroup.new(model)
+partitionGroup.setSpace(sub_surface.space.get)
+
+points = OpenStudio::Point3dVector.new
+points << OpenStudio::Point3d.new(0, 1, 0.3)
+points << OpenStudio::Point3d.new(0, 0, 0.3)
+points << OpenStudio::Point3d.new(1, 0, 0.3)
+points << OpenStudio::Point3d.new(1, 1, 0.3)
+inside_shelf = OpenStudio::Model::InteriorPartitionSurface.new(points, model)
+inside_shelf.setInteriorPartitionSurfaceGroup(partitionGroup)
+
+#### Outside Shelf
+# This shading sf must have a construction as  it's used as an outside shelf
+shadingSurface = sub_surface.addOverhangByProjectionFactor(0.5, 0.1).get
+shadingMat = OpenStudio::Model::StandardOpaqueMaterial.new(model)
+shadingMat.setName('C12 - 2 IN HW CONCRETE - PAINTED WHITE')
+shadingMat.setRoughness('MediumRough')
+shadingMat.setThickness(OpenStudio.convert(2, 'in', 'm').get)
+shadingMat.setConductivity(1.729577)
+shadingMat.setDensity(2242.585)
+shadingMat.setSpecificHeat(836.8)
+shadingMat.setThermalAbsorptance(0.9)
+shadingMat.setSolarAbsorptance(0.3)
+shadingMat.setVisibleAbsorptance(0.3)
+
+shadingConstruction = OpenStudio::Model::Construction.new([shadingMat])
+shadingConstruction.setName('Outside Shelf Construction')
+
+shadingSurface.setConstruction(shadingConstruction)
```

Comment:
Outside shelf is expected to be a Shading Surface, and that shading surface must have a construction assigned.